### PR TITLE
MGDCTRS-1745: Enabling Label Selection feature in Camel-K

### PR DESCRIPTION
**Pull request by shivamG640**
This PR is against https://issues.redhat.com/browse/MGDCTRS-1745, where we are basically enabling the Label Selection feature.

### CODE REVIEW

**Review by valdar — APPROVED** [review]

File: etc/kubernetes/manifests/base/apps/camel-k/patches/patch-camel-k-deployment.yaml [hunk 1]
```diff
@@ -21,6 +21,8 @@ spec:
                 name: camel-k-override-config
                 key: log.level
                 optional: true
+          - name: OPERATOR_ID
```
Comment (valdar) [inline etc/kubernetes/manifests/base/apps/camel-k/patches/patch-camel-k-deployment.yaml:24]: maybe the above comments on line 14 and 15 can be removed?
`          #- name: KAMEL_OPERATOR_ID            
          #  value: "cos-fleetshard-operator-camel"`

**Review by valdar — CHANGES_REQUESTED** [review]

File: etc/kubernetes/manifests/base/apps/camel-k/patches/patch-camel-k-deployment.yaml (hunk 1 above)
Comment (shivamG640) [reply etc/kubernetes/manifests/base/apps/camel-k/patches/patch-camel-k-deployment.yaml:22]: Why do we need KAMEL_OPERATOR_ID env var?

Comment (valdar) [reply etc/kubernetes/manifests/base/apps/camel-k/patches/patch-camel-k-deployment.yaml:22]: We don't, that is why those lines can be deleted. It was an old name for the OPERATOR_ID env var

Comment (shivamG640) [reply etc/kubernetes/manifests/base/apps/camel-k/patches/patch-camel-k-deployment.yaml:22]: Ah, I already deleted them! Thnaks

File: etc/kubernetes/manifests/base/apps/cos-fleetshard-operator-camel/app-config/application.properties [hunk 2]
```diff
@@ -5,5 +5,5 @@ quarkus.log.category."org.bf2.cos.fleetshard.sync".level = DEBUG
 quarkus.log.category."org.bf2.cos.fleetshard.client".level = DEBUG
 quarkus.log.category."io.fabric8.kubernetes.client.internal.VersionUsageUtils".level = ERROR
 
-cos.operator.camel.label-selection.enabled = false
+cos.operator.camel.label-selection.enabled = true
```
Comment (rinaldodev) [inline etc/kubernetes/manifests/base/apps/cos-fleetshard-operator-camel/app-config/application.properties:8]: Is the value `cos-fleetshard-operator-camel` configurable? Should we make it in a future PR? Just a question/suggestion. LGTM.

File: etc/kubernetes/manifests/overlays/local/camel-k/integration-platform.yaml [hunk 3]
```diff
@@ -1,4 +1,6 @@
 apiVersion: camel.apache.org/v1
 kind: IntegrationPlatform
 metadata:
-  name: camel-k
+  name: cos-fleetshard-operator-camel
+  annotations:
+    camel.apache.org/operator.id: "cos-fleetshard-operator-camel"
```
Comment (valdar) [inline etc/kubernetes/manifests/overlays/local/camel-k/integration-platform.yaml:6]: @rinaldodev are these overlays enough to make the changes in prod as well? once the addon generation picks this up?

**Review by valdar — CHANGES_REQUESTED** [review]

File: etc/kubernetes/manifests/overlays/local/camel-k/integration-platform.yaml (hunk 3 above)
Comment (rinaldodev) [reply etc/kubernetes/manifests/overlays/local/camel-k/integration-platform.yaml:6]: Yes. The staging overlay is used for prod as well.

**Review by valdar — APPROVED** [review]

**Pull request merged**